src/sickle.ts

```diff
@@ -380,7 +384,14 @@ class Annotator {
     try {
       return getJSDocAnnotation(comment);
     } catch (e) {
-      this.fail(e.toString());
+      this.diagnostics.push({
+        file: this.file,
+        start: node.getFullStart() + pos,
+        length: end - pos,
+        messageText: e.message,
+        category: ts.DiagnosticCategory.Error,
+        code: 0,
```

Comment:
How about having Sickle.Diagnostic interface instead of reusing ts.Diagnostic, with somewhat unclear codes?